### PR TITLE
Replace template<class> with template<typename>

### PR DESCRIPTION
**Pull request by matssson**
Since the book focuses heavily on types, I think that the C++ keyword typename is more appropriate, clear and avoids confusion for those less experienced with C++, and I think it's generally good practice to use typename instead of the overloaded meaning of class.

I also added a missing step in the composition argument for fmap Nothing and fixed some whitespace formatting.
There is trailing whitespace in the licence file and in errata-1.0.0.md, but I didn't change them since I think it's more interesting for the readers to know when they were last updated properly.

### CODE REVIEW

File: src/content/1.7/functors.tex [hunk 1]
```diff
@@ -206,6 +206,8 @@ \subsection{Equational Reasoning}
   fmap g Nothing
 = { definition of fmap }
   fmap g (fmap f Nothing)
+= { definition of composition }
```
Comment (drupol) [inline src/content/1.7/functors.tex:209]: This snippet seems to be out of scope of this PR, is it intended ?

**Comment by drupol** [comment]
Dear @matssson,

Do you plan to implement the changes in this PR ?

Thanks

File: src/content/1.7/functors.tex (hunk 1 above)
Comment (matssson) [reply src/content/1.7/functors.tex:209]: It was intended, but you're right that it's out of scope - I'll remove it.

Also thanks for the reminder, I completely forgot about this PR. I'll fix it this week.

Comment (drupol) [reply src/content/1.7/functors.tex:209]: no problem :)

**Comment by matssson** [comment]
I fixed the changes now in a new PR

https://github.com/hmemcpy/milewski-ctfp-pdf/pull/309

I redid the trailing whitespace part since there were a lot of changes since the original PR.
